oppija.py

```diff
@@ -34,20 +45,73 @@ def opisCountHandler(self, bot, update, args=''):
         bot.sendMessage(chat_id=update.message.chat_id, text=(str(result[0]) + ' opis'))
 
-    def randomOppiHandler(self, bot, update):
-        result = db.randomOppi(update.message.chat.id)
-        bot.sendMessage(chat_id=update.message.chat_id, text=(result[0] + ': ' + result[1]))
+    def randomOppiHandler(self, bot, update, inverted=False):
+        if (inverted):
+            result = db.randomOppi(update.message.chat.id)
+            inverted_result = self.invertStringList(result)
+            bot.sendMessage(chat_id=update.message.chat_id, text=(inverted_result[1] + ' :' + inverted_result[0]))
+        else:
+            result = db.randomOppi(update.message.chat.id)
+            bot.sendMessage(chat_id=update.message.chat_id, text=(result[0] + ': ' + result[1]))
+
+    def invertStringList(self, list):
+        # Reference table for the Unicode chars: http://www.upsidedowntext.com/unicode
+        chars_standard = 'abcdefghijklmnopqrstuvwxyzåäö'
+        chars_inverted = 'ɐqɔpǝɟbɥıɾʞןɯuodbɹsʇnʌʍxʎzɐɐo'
+
+        chars_standard += '_,;.?!/\\\'<>(){}[]`&'
+        chars_inverted += '‾\'؛˙¿¡/\\,><)(}{][,⅋'
+
+        chars_standard += 'ABCDEFGHIJKLMNOPQRSTUVWXYZÅÄÖ'
+        chars_inverted += '∀qϽᗡƎℲƃHIſʞ˥WNOԀὉᴚS⊥∩ΛMXʎZ∀∀O'
+
+        chars_standard += '0123456789'
+        chars_inverted += '0ƖᄅƐㄣϛ9ㄥ86'
+
+        inverted_list = []
+        for string in list:
+            inverted_string = ''
+
+            for char in string:
+                try:
+                    charIndex = chars_standard.index(char)
+                except:
+                    inverted_string += char
+                    continue
+                inverted_string += chars_inverted[charIndex]
+
+            # Reverse the string to make it readable upside down
+            inverted_list.append(inverted_string[::-1])
+
+        return inverted_list
+
+    def jokotaiHandler(self, bot, update, args=''):
+        sides = ['kruuna', 'klaava']
+        maximalRigging = random.choice(sides)
+        riggedQuestion = re.match(r"^(\?\?)\s(\S+)$", "?? " + maximalRigging)
+
+        bot.sendMessage(chat_id=update.message.chat_id, parse_mode='Markdown', text='*♪ Se on kuulkaas joko tai, joko tai! ♪*')
+        self.defineTerm(bot, update, riggedQuestion)
 
     def messageHandler(self, bot, update):
         msg = update.message
         if msg.text is not None:
-            # Matches messages in format "?? something"
+            # Matches messages in formats "?? something" and "¿¿ something"
             question = re.match(r"^(\?\?)\s(\S+)$", msg.text)
+            inverted_question = re.match(r"^(\¿\¿)\s(\S+)$", msg.text)
             if question:
                 self.defineTerm(bot, update, question)
+            elif inverted_question:
+                self.defineTerm(bot, update, inverted_question, True)
+
             # Matches message "?!"
             elif re.match(r"^(\?\!)$", msg.text):
                 self.randomOppiHandler(bot, update)
+
```

Comment:
Shouldn't this be ¿¡ ?

Reply:
No, the order is intentional. If you make a long Oppi, use this command, flip your screen upside down, and you'll see why. ;)